bot/exts/evergreen/issues.py

```diff
@@ -159,79 +199,74 @@ async def issue(
+        # Remove duplicates
+        numbers = set(numbers)
 
-        if result == FetchIssueErrors.value_error:
-            await ctx.invoke(self.bot.get_command('help'), 'issue')
-
-        elif result == FetchIssueErrors.max_requests:
+        if len(numbers) > MAXIMUM_ISSUES:
             embed = discord.Embed(
                 title=random.choice(ERROR_REPLIES),
                 color=Colours.soft_red,
-                description=f"Too many issues/PRs! (maximum of {MAX_REQUESTS})"
+                description=f"Too many issues/PRs! (maximum of {MAXIMUM_ISSUES})"
             )
             await ctx.send(embed=embed)
+            await invoke_help_command(ctx)
 
-        elif isinstance(result, list):
-            # Issue/PR format: emoji to show if open/closed/merged, number and the title as a singular link.
-            resp = self.get_embed(result, user, repository)
-            await ctx.send(embed=resp)
-
-        elif isinstance(result, str):
-            await ctx.send(result)
+        results = [await self.fetch_issues(number, repository, user) for number in numbers]
+        await ctx.send(embed=self.format_embed(results, user, repository))
 
     @commands.Cog.listener()
     async def on_message(self, message: discord.Message) -> None:
         """Command to retrieve issue(s) from a GitHub repository using automatic linking if matching <repo>#<issue>."""
-        # Ignore messages not in whitelisted categories / channels, only when in guild.
-        if message.guild and not (
-            message.channel.category.id in WHITELISTED_CATEGORIES
-            or message.channel.id in WHITELISTED_CHANNELS_ON_MESSAGE
-        ):
+        if not self.repo_regex:
+            log.warning("repo_regex isn't ready, cannot look for issues.")
+            return
+
+        # Ignore bots
+        if message.author.bot:
             return
 
-        message_repo_issue_map = re.findall(fr"({self.repo_regex})#(\d+)", message.content)
+        # `issues` will hold a list of two element tuples `(repository, issue_number)`
+        issues = re.findall(fr"\b({self.repo_regex})#(\d+)\b", message.content)
         links = []
 
-        if message_repo_issue_map:
+        if issues:
+            # Block this from working in DMs
             if not message.guild:
                 await message.channel.send(
                     embed=discord.Embed(
                         title=random.choice(NEGATIVE_REPLIES),
                         description=(
-                            "You can't retreive issues from DMs. "
+                            "You can't retrieve issues from DMs. "
                             f"Try again in <#{Channels.community_bot_commands}>"
                         ),
                         colour=discord.Colour.red()
                     )
                 )
                 return
-            for repo_issue in message_repo_issue_map:
+
+            log.trace(f"Found {issues = }")
```

Comment:
Include the message ID here, otherwise this statement is of no use.

Reply:
This is just a trace statement, it was just for me to understand what this variable would contain. It will only be shown in development it is pretty trivial to know which message triggered it.